src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -110,9 +110,7 @@ public function last(string $stampFqcn): ?StampInterface
      *
      * @param class-string<TStamp>|null $stampFqcn
      *
-     * @return StampInterface[]|StampInterface[][] The stamps for the specified FQCN, or all stamps by their class name
-     *
-     * @psalm-return ($stampFqcn is string : array<class-string<StampInterface>, list<StampInterface>> ? list<TStamp>)
+     * @return ($stampFqcn is string ? list<TStamp> : array<class-string<StampInterface>, list<StampInterface>>) The stamps for the specified FQCN, or all stamps by their class name
```

Comment:
why merge return and psalm-return? I think we didn't anywhere yet so we shouldn't start in a PR like that

Reply:
i thought i made psalm trip, but no, it still errors. We should template stamps property still, but that's not a public API concern. Also i was unsure how to do it :sweat_smile: 

in fact we use both conditional `@return` and `@psalm-return`. Merge across the board welcome i think.

https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code
https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40psalm-return+%28%22&type=code

Reply:
I don't think we use conditional types in `@return`

Reply:
we do https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code

Reply:
we do https://github.com/search?q=repo%3Asymfony%2Fsymfony+%22%40return+%28%22&type=code

Reply:
i dont mind, you can cherry pick a2c660b94410f4abf54e79b932702a3172ce8a9f